src/scala/org/pantsbuild/zinc/compiler/Main.scala

```diff
@@ -187,18 +197,87 @@ object Main {
     Settings.SettingsParser
       .parse(preprocessArgs(context.getArgs), Settings()) match {
       case Some(settings) =>
+        val workingDirectory = new File(context.getWorkingDirectory)
         mainImpl(
-          settings.withAbsolutePaths(new File(context.getWorkingDirectory)),
+          settings.withAbsolutePaths(workingDirectory),
           startTime,
-          n => context.exit(n))
+          n => context.exit(n),
+          workingDirectory
+        )
       case None => {
         println("See zinc-compiler --help for information about options")
         context.exit(1)
       }
     }
   }
 
-  def mainImpl(settings: Settings, startTime: Long, exit: Int => Unit): Unit = {
+  def toLsp(severity: Severity): DiagnosticSeverity = {
+    severity match {
+      case Severity.Error => DiagnosticSeverity.Error
+      case Severity.Warn  => DiagnosticSeverity.Warning
+      case Severity.Info  => DiagnosticSeverity.Information
+      // Note: DiagnosticSeverity also has a Hint level, but we won't use it here as xsbti doesn't go that far
+    }
+  }
+  def toZeroBased(x: Int): Int = {
```

Comment:
If you want to get really fancy:
```
implicit class ZeroBasedInt(x: Int) { def zeroBased() { x - 1 } }

// Usage:
2.zeroBased() // 1
val x = 3; x.zeroBased() // 2
```

Docs: https://docs.scala-lang.org/overviews/core/implicit-classes.html

Not suggesting you do this, but if you find it useful, I wouldn't be opposed.

Reply:
Same with `toLSP`

Reply:
I think I'll pass for now because it's purely stylistic and I find the pure function a little bit more straightforward. Maybe as I write more scala. I'll develop a taste for it and change my mind in future commits :smile: